legit.py

```diff
@@ -0,0 +1,54 @@
+from git import Repo
+from git.exc import GitCommandError
+from typing import Iterable
+
+
+class Repository:
+
+    def __init__(self, path: str) -> None:
+        self.repo = Repo.init(path)
+
+    @classmethod
+    def clone(cls, url: str, path: str) -> str:
+        try:
+            Repo.clone_from(url, path)
+        except GitCommandError:
+            return None
```

Comment:
The semantics of quashing errors creating something and returning None/empty string/whatever instead are good when it makes business sense in the context of the call. E.g. `'ewqfqwef'[50:]` evaluates to an empty string because you are asking for a slice that is impossible and thus empty. I don't think here is a good place to quash errors, a try/except block in the user's code would be much clearer than a call immediately followed by an `if` checking for None.

Reply:
I don't get it, please, could you explain to me (and possibly to others) what this func supposed to return and why one more time?